### PR TITLE
Fix Allowed Intent Input Type

### PR DESCRIPTION
**Pull request by ritvikshrivastava**
The current docs mention the use of a list whereas internal validation is for tuple. Added list to the accepted type for passing in allowed intents.

### CODE REVIEW

**Comment by minhtuev** [comment]
Should we update the logic to use both list and tuple?

**Comment by ritvikshrivastava** [comment]
@minhtuev wasn't the idea to keep it a tuple to ensure immutability? I mean it can be converted internally, yeah.

**Comment by serapio** [comment]
It seems like what we actually want is to leave `list` in the documentation for simplicity, and pass `converter=tuple` when constructing it. 
https://github.com/cisco/mindmeld/blob/2178a92acaa3008f7ef7ce80e0eaf319c3b1da46/mindmeld/components/request.py#L183

File: tests/components/test_nlp.py [hunk 1]
```diff
@@ -617,7 +617,7 @@ def test_sys_entity_feature(kwik_e_mart_nlp):
 
 
 test_data_dyn = [
-    ("kadubeesanahalli", None, "store_info", "", ""),
+    ("kadubeesanahalli", None, "store_info", "get_store_hours", ""),
```
Comment (vijay120) [inline tests/components/test_nlp.py:620]: This test change seems unrelated to the code change.

Comment (ritvikshrivastava) [reply tests/components/test_nlp.py:620]: It's in reference to a failing test for all PRs (refer to Minh's comment above). Should I move it to a separate PR?
